src/speculators/data_generation/fp8_hidden_states_connector.py

```
@@ -0,0 +1,80 @@
+
+if TYPE_CHECKING:
+    from vllm.v1.attention.backend import AttentionMetadata
+
+
+class FP8HiddenStatesConnector(_eh_mod.ExampleHiddenStatesConnector):
+    """Quantizes hidden states to float8_e4m3fn with per-token scaling
+    before saving to safetensors.
+
+    The output file contains three tensors::
+
+        hidden_states        – fp8 [seq_len, num_layers, hidden_size]
```

Ruff (0.15.12)

[warning] Docstring contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF002)